src/JustSaying/Messaging/Middleware/MiddlewareBase`2.cs

```diff
@@ -10,6 +10,8 @@ public abstract class MiddlewareBase<TContext, TOut>
         return this;
     }
 
+    public bool HasNext => _next != null;
```

Comment:
I wonder if this should just be internal?

Reply:
Ah yeah this is fair, good point.